src/Essentials/src/MediaGallery/MediaGallery.android.cs

```diff
@@ -0,0 +1,329 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Content.PM;
+using Android.Provider;
+using Android.Webkit;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Storage;
+using Environment = Android.OS.Environment;
+using File = Java.IO.File;
+using Path = System.IO.Path;
+using Stream = System.IO.Stream;
+using Uri = Android.Net.Uri;
+#if ANDROID30_0_OR_GREATER
+using MediaColumns = Android.Provider.MediaStore.IMediaColumns;
+#elif MONOANDROID10_0
+global using MediaColumns = Android.Provider.MediaStore.MediaColumns;
+#endif
+
+namespace Microsoft.Maui.Media
+{
+	partial class MediaGalleryImplementation : IMediaGallery
+	{
+		static TaskCompletionSource<(Intent, Result)> TcsPick;
+		static TaskCompletionSource<(Intent, Result)> TcsCamera;
+
+		public bool IsSupported => OperatingSystem.IsAndroidVersionAtLeast(21);
+
+		public bool CheckCaptureSupport(MediaFileType type)
+		{
+			if (!IsSupported || !(Platform.AppContext?.PackageManager?.HasSystemFeature(PackageManager.FeatureCameraAny) ?? false))
+				return false;
+			using var intent = GetCameraIntent(type);
+			return PlatformUtils.IsIntentSupported(intent);
+		}
+
+		public async Task<IEnumerable<MediaFileResult>> PlatformCaptureAsync(MediaFileType type, CancellationToken token = default)
+		{
+			token.ThrowIfCancellationRequested();
+			Intent intent = null;
+			Uri outputUri = null;
+
+			try
+			{
+				TcsCamera = new TaskCompletionSource<(Intent, Result)>(TaskCreationOptions.RunContinuationsAsynchronously);
+				intent = GetCameraIntent(type);
+
+				var fileName = $"{GetNewImageName()}.jpg";
+				var tempFilePath = GetFilePath(fileName);
+				using var file = new File(tempFilePath);
+				if (!file.Exists())
+					file.CreateNewFile();
+				outputUri = FileProvider.GetUriForFile(file);
+				intent.PutExtra(MediaStore.ExtraOutput, outputUri);
+
+				CancelTaskIfRequested(token, TcsCamera);
+
+				StartActivity(intent, PlatformUtils.requestCodeMediaCapture, token, TcsCamera);
+
+				CancelTaskIfRequested(token, TcsCamera, false);
+				var result = await TcsCamera.Task.ConfigureAwait(false);
+				if (result.Item2 == Result.Ok)
+					return new[] { new MediaFileResult(fileName, outputUri, tempFilePath) };
+
+				outputUri?.Dispose();
+				return null;
```

Comment:
Returning null gut the method signature is Task<IEnumerable<MediaFileResult>>, either change the signature or return Array.Empty<MediaFileResult>()